### DISCUSSION TITLE
camera_helper.start size

### DISCUSSION THREAD

**Discussion by JackyCA** [j20001970/GDMP#33] [Q&A]
Hello. 

I have been trying to use GDMP demo for my project. I realize that `camera_helper.start`'s second argument does not always translate to the final image size, as accessible from the `update_image` function, on the line of `image.get_size()` more specifically. 

That is to some extent expected because the models can only take on fixed input dimensions and the camera has certain hardware limitations. The problem is on my PC I have a few dimensions to choose from, e.g. (320, 240), (800, 600), but after exporting to my Android device, I am always stuck with (1088, 1088) regardless of my input. So I have two questions

1. How can I change the image size on Android?
2. Does the high image resolution actually have an impact on the app's performance?

P.S. I am using the demo hand landmarker for 4.1.1. If new releases have already fixed this issue, just let me know. 

Thank you.

**Comment by j20001970** [comment]
Have you tried other camera sizes? You can also try rotating the size to see if it make the difference, for example (240, 320), (600, 800), (640, 480) and (480, 640). The current state of MediaPipeCameraHelper is admittedly inconsistent across different platforms.

1. Camera frame size on Android is determined by `Preview.Builder.setTargetResolution` from CameraX, calling `camera_helper.start` the second parameter `size` is passed to Java class `GDMPCameraHelper.startCamera` through JNI. So changing the `size` argument should also change camera frame size, in theory at least.
2. In general yes, while MediaPipe tasks take fixed size image for inference, generating camera frames on Android involves GLES texture operations, which may require more GPU resources for larger frame size. And if you would like to display camera frames on Godot, converting GPU images to CPU-accessible pixel data is required, displaying a large image alone would be a bottleneck for your apps.

**Reply by JackyCA** [reply]
I have tried many sizes and rotating the size. It does change the camera size on my Windows PC, not on my Samsung Tablet though. I guess I will have to bear with the somewhat low FPS for now. 

Thanks anyway for the swift reply.
